spyder/config/main.py

```diff
@@ -426,7 +427,7 @@
               'editor/run cell and advance': RUN_CELL_AND_ADVANCE_SHORTCUT,
               # -- In plugins/editor.py
               'editor/show/hide outline': "Ctrl+Alt+O",
-              'editor/show/hide project explorer': "Ctrl+Alt+P",

```

Comment:
Well, the Project Explorer can be closed at any time, so it's not enforced on users. It's just a visual clue that you're working on a project.

>  I find it pretty annoying that it always comes back when I launch Spyder

Please submit a PR to fix that. I don't have a problem with it :-)